services/src/main/scala/cromwell/services/metadata/impl/pubsub/PubSubMetadataServiceActor.scala

```diff
@@ -0,0 +1,133 @@
+import cromwell.cloudsupport.gcp.GoogleConfiguration
+import cromwell.cloudsupport.gcp.auth.ServiceAccountMode
+import cromwell.cloudsupport.gcp.auth.ServiceAccountMode.PemFileFormat
+import cromwell.core.Dispatcher._
+import cromwell.services.metadata._
+import cromwell.services.metadata.MetadataService.{MetadataWriteFailure, MetadataWriteSuccess, PutMetadataAction, PutMetadataActionAndRespond}
+import net.ceedubs.ficus.Ficus._
+import org.broadinstitute.dsde.workbench.google.{GooglePubSubDAO, HttpGooglePubSubDAO}
+import spray.json._
+import cats.instances.future._
+import cats.syntax.functor._
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+/**
+  * A *write-only* metadata service implementation which pushes all events to Google PubSub. The expectation is that
+  * metadata reads are being handled outside of this Cromwell instance.
+  *
+  * For now it requires a google service account auth with a PEM file. If the underlying PubSub library is updated to
+  * allow for other forms of auth this becomes viable for us. Note that currently Cromwell complains that PEM is
+  * deprecated, so we might need to contribute a patch to workbench-lib's pubsub tooling.
+  *
+  * This is intentionally not documented, considering there exists no way to consume/read metadata if this is active
+  * anyone using this is by definition more advanced than the Cromwell developers
+  *
+  * Potential TODOs to consider if moving this forward from proof-of-concept:
+  *  - This will probably need some measure of batching, need to see it in action before proceeding down that path
+  *  - Graceful shutdown could also be useful. The pubsub calls will autoretry in the face of transient errors which
+  *     means there might be stuff hanging around on a shutdown request.
+  *  - Currently service actors which fail to instantiate properly don't bring down Cromwell. It's easier to screw
+  *     it up here, we might want to revisit that strategy
+  *  - Revisit the Alpakka connector instead of the workbench-lib version. There was a bug they fixed but hadn't released
+  *     it yet. Also I'm pretty sure that same bug was repeated in a few other places.
+  */
+class PubSubMetadataServiceActor(serviceConfig: Config, globalConfig: Config) extends Actor with ActorLogging {
+  implicit val ec = context.dispatcher
+
+  // The auth *must* be a service account auth but it might not be named service-account.
+  val googleAuthName = serviceConfig.as[Option[String]]("auth").getOrElse("service-account")
+
+  val googleProject = serviceConfig.as[String]("project")
+  val pubSubTopicName = serviceConfig.as[Option[String]]("topic").getOrElse("cromwell-metadata")
+  val pubSubSubscriptionName = serviceConfig.as[Option[String]]("subscription")
+  val pubSubAppName = serviceConfig.as[Option[String]]("appName").getOrElse("cromwell")
+
+  val pubSubConnection = createPubSubConnection()
+  createTopicAndSubscription().failed foreach { e => throw e }
```

Comment:
Sorry to beat this horse but this side effecty stuff should be done elsewhere

Reply:
Where? 

This kills the actor which bubbles up the supervision chain and ultimately stops Cromwell from loading.

Reply:
Although today that doesn't seem to be the case, need to figure that out